test/test_spectral_ops.py

```diff
@@ -284,13 +319,17 @@ def test_fft_round_trip(self, device, dtype):
                 }
 
                 y = backward(forward(x, **kwargs), **kwargs)
+                if x.dtype is torch.half and y.dtype is torch.complex32:
+                    # Since type promotion currently doesn't work with complex32
```

Comment:
we should just add a small PR to add type promotion support for complex32. it should be fairly straightforward -- you just need to update the type promotion table

Reply:
Noted.